src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -520,7 +520,8 @@ private void AddTupleTypeName(INamedTypeSymbol symbol)
                 }
 
                 VisitFieldType(element);
-                if (!element.IsImplicitlyDeclared)
+                if (!element.IsImplicitlyDeclared &&
+                    !(symbol.IsGenericType && symbol.IsDefinition)) // C# tuples are generic
```

Comment:
>!(symbol.IsGenericType && symbol.IsDefinition) [](start = 20, length = 46)

This logic looks suspicious. Is this going to do the right thing for VB symbols? Is this PR affecting behavior of ```IsImplicitlyDeclared```? Is implementation of VB SymbolDisplay going to do the right thing for C# symbols now, I do not see any changes made there? 
Also, I am assuming that for ```(T1 Item1, T2)``` we want to print "Item1", but for (T1, T2 Item2) we don't want to print it. I don't see how ```(symbol.IsGenericType && symbol.IsDefinition)``` could yield different value for both these types. Perhaps I am misunderstanding the scenario.

Reply:
In this PR, we're not changing the behavior of `IsImplicitlyDeclared` on `TupleElementFieldSymbol` (wrapped fields) or `TupleErrorFieldSymbol`. But for fields that are now unwrapped, namely fields that appear in the `ValueTuple<...>` types, `IsImplicitlyDeclared` now returns `false`.
For instance, in  `struct ValueTuple<T1, T2) { T1 Item1; T2 Item2; string goo; }` the three fields have `IsImplicitlyDeclared=false`. But when we display such a type (which is a definition), we don't want to display `(T1 Item1, T2 Item2)` but rather `(T1, T2)`. We can distinguish that using `IsDefinition`.
But in VB, all tuple types are definitions, so I'm using the fact that they are not generic to avoid affecting them.

I'll look for VB tests to make sure the scenario you raised is covered.

---
In reply to: [426205983](https://github.com/dotnet/roslyn/pull/44231#discussion_r426205983) [](ancestors = 426205983)

Reply:
It feels to me that relying on IsImplicitlyDeclared to decide whether to include names is fragile and probably a wrong thing to do. It feels like that should be decided on whether explicit names were specified in the tuple type syntax used to "create" the tuple type. That is not the same as IsImplicitlyDeclared, or at least it is no longer the same. We probably should base the logic on a more specific API/information.

---
In reply to: [426687365](https://github.com/dotnet/roslyn/pull/44231#discussion_r426687365) [](ancestors = 426687365,426205983)

Reply:
It might be worth discussing this issue in more depth offline before we decide the course of action

---
In reply to: [426774418](https://github.com/dotnet/roslyn/pull/44231#discussion_r426774418) [](ancestors = 426774418,426687365,426205983)